src/datasets/utils/file_utils.py

```diff
@@ -1159,40 +1159,34 @@ def _prepare_single_hop_path_and_storage_options(
         }
     else:
         storage_options = {}
-    if storage_options:
-        storage_options = {protocol: storage_options}
-    if protocol in ["http", "https"]:
-        storage_options[protocol] = {
-            "headers": {
-                **get_authentication_headers_for_url(urlpath, token=token),
-                "user-agent": get_datasets_user_agent(),
-            },
-            "client_kwargs": {"trust_env": True},  # Enable reading proxy env variables.
-            **(storage_options.get(protocol, {})),
-        }
+    if protocol in {"http", "https"}:
+        client_kwargs = storage_options.setdefault("client_kwargs", {})
```

Comment:
you might want to modify a copy of the user's `storage_options`

Reply:
Definitely! Thanks for the insight! :sweat_smile:

Reply:
Done.